staging/src/k8s.io/component-base/metrics/legacyregistry/registry.go

```diff
@@ -44,10 +45,8 @@ var (
 )
 
 func init() {
-	//nolint:staticcheck // SA1019 - replacement function still calls prometheus.NewProcessCollector().
-	RawMustRegister(prometheus.NewProcessCollector(prometheus.ProcessCollectorOpts{}))
-	//nolint:staticcheck // SA1019 - replacement function still calls prometheus.NewGoCollector().
-	RawMustRegister(prometheus.NewGoCollector())
+	RawMustRegister(collectors.NewProcessCollector(collectors.ProcessCollectorOpts{}))
+	RawMustRegister(collectors.NewGoCollector(collectors.WithGoCollections(collectors.GoRuntimeMetricsCollection)))
```

Comment:
collectors.GoRuntimeMetricsCollection will be deprecated once we upgrade client_golang to 1.13.0, so we might aswell prefer upgrading to the new version of client_golang and use the new approach to enable the runtime/metrics package.

More about that in: https://github.com/prometheus/client_golang/blob/release-1.13/prometheus/collectors/go_collector_latest.go#L135-L138

Reply:
Sure, that makes perfect sense.

I opened https://github.com/kubernetes/kubernetes/pull/112052 for client_golang bump to 1.13.0, will rebase and adjust the collector registration once the dependency update PR merges.

Reply:
112052 got submitted, I rebased the PR and changed the collector's options. Done.